Sources/StreamChatUI/ImageGalleryVC.swift

```diff
@@ -0,0 +1,232 @@
+        
+        closeButton.addTarget(self, action: #selector(closeButtonTapped), for: .touchUpInside)
+        shareButton.addTarget(self, action: #selector(shareButtonTapped), for: .touchUpInside)
+    }
+    
+    override open func setUpLayout() {
+        super.setUpLayout()
+        
+        view.embed(attachmentsCollectionView)
+        
+        view.addSubview(topBarView)
+        topBarView.pin(anchors: [.leading, .trailing, .top], to: view)
+        
+        let topBarContainerStackView = ContainerStackView()
+            .withoutAutoresizingMaskConstraints
+        topBarView.embed(topBarContainerStackView)
+        topBarContainerStackView.preservesSuperviewLayoutMargins = true
+        topBarContainerStackView.isLayoutMarginsRelativeArrangement = true
+        
+        topBarContainerStackView.addArrangedSubview(closeButton)
+        
+        let infoContainerStackView = ContainerStackView()
+        infoContainerStackView.axis = .vertical
+        infoContainerStackView.alignment = .center
+        topBarContainerStackView.addArrangedSubview(infoContainerStackView)
+        infoContainerStackView.pin(anchors: [.centerX], to: view)
+        
+        userLabel.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        infoContainerStackView.addArrangedSubview(userLabel)
+        
+        topBarContainerStackView.addArrangedSubview(UIView.spacer(axis: .horizontal))
+        
+        view.addSubview(bottomBarView)
+        bottomBarView.pin(anchors: [.leading, .trailing, .bottom], to: view)
+        
+        let bottomBarContainerStackView = ContainerStackView()
+            .withoutAutoresizingMaskConstraints
+        bottomBarContainerStackView.preservesSuperviewLayoutMargins = true
+        bottomBarContainerStackView.isLayoutMarginsRelativeArrangement = true
+        bottomBarView.embed(bottomBarContainerStackView)
+        
+        shareButton.setContentHuggingPriority(.streamRequire, for: .horizontal)
+        bottomBarContainerStackView.addArrangedSubview(shareButton)
+
+        currentPhotoLabel.setContentHuggingPriority(.defaultLow, for: .horizontal)
+        bottomBarContainerStackView.addArrangedSubview(currentPhotoLabel)
+        currentPhotoLabel.pin(anchors: [.centerX], to: view)
+        
+        bottomBarContainerStackView.addArrangedSubview(.spacer(axis: .horizontal))
+    }
+    
+    override open func updateContent() {
+        super.updateContent()
+        
+        images = content.attachments
+            .filter { $0.type == .image }
+            .compactMap { $0 as? ChatMessageDefaultAttachment }
+        
+        userLabel.text = content.author.name
+        
+        currentPhotoLabel.text = L10n.currentSelection(currentPage + 1, images.count)
+    }
+    
+    @objc
+    open func closeButtonTapped() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @objc
+    open func shareButtonTapped() {
+        guard let imageURL = images[currentPage].imageURL else { return }
```

Comment:
do we want to share the image URL or the image itself? Sharing the image itself _might_ be a little trickier to do, but possible